controllers/operator/manager.go

```diff
@@ -402,38 +402,40 @@ func (m *ODLMOperator) GetOperatorNamespace(installMode, namespace string) strin
 // GetOperandFromRegistry gets the Operand from the OperandRegistry
 func (m *ODLMOperator) GetOperandFromRegistry(ctx context.Context, reg *apiv1alpha1.OperandRegistry, operandName string) (*apiv1alpha1.Operator, error) {
 	opt := reg.GetOperator(operandName)
-	// Get excluded CatalogSource from annotation
-	// excluded-catalogsource: catalogsource1, catalogsource2
-	var excludedCatalogSources []string
-	if reg.Annotations != nil && reg.Annotations["excluded-catalogsource"] != "" {
-		excludedCatalogSources = strings.Split(reg.Annotations["excluded-catalogsource"], ",")
-	}
-	// Get catalog used by ODLM itself by check its own subscription
-	opts := []client.ListOption{
-		client.MatchingLabels{fmt.Sprintf("operators.coreos.com/ibm-odlm.%s", util.GetOperatorNamespace()): ""},
-		client.InNamespace(util.GetOperatorNamespace()),
-	}
-	odlmCatalog := ""
-	odlmCatalogNs := ""
-	odlmSubList := &olmv1alpha1.SubscriptionList{}
-	if err := m.Reader.List(ctx, odlmSubList, opts...); err != nil || len(odlmSubList.Items) == 0 {
-		klog.Warningf("No Subscription found for ibm-odlm in the namespace %s", util.GetOperatorNamespace())
-	} else {
-		odlmCatalog = odlmSubList.Items[0].Spec.CatalogSource
-		odlmCatalogNs = odlmSubList.Items[0].Spec.CatalogSourceNamespace
-	}
-
-	if opt.SourceName == "" || opt.SourceNamespace == "" {
-		catalogSourceName, catalogSourceNs, err := m.GetCatalogSourceFromPackage(ctx, opt.PackageName, opt.Namespace, opt.Channel, reg.Namespace, odlmCatalog, odlmCatalogNs, excludedCatalogSources)
-		if err != nil {
-			return nil, err
-		}
+	if opt != nil {
```

Comment:
suggest to invert the condition, so that if opt is nil, then return nil. Will be easier to read the code, and best practice is to return as early as possible.